crates/vhost-device-gpio/src/backend.rs

```diff
@@ -214,33 +221,41 @@ fn start_backend(args: GpioArgs) -> Result<()> {
         let socket = config.socket_path.to_owned() + &i.to_string();
         let cfg = config.devices.inner[i];
 
-        let handle: JoinHandle<Result<()>> = spawn(move || loop {
-            // A separate thread is spawned for each socket and can connect to a separate guest.
-            // These are run in an infinite loop to not require the daemon to be restarted once a
-            // guest exits.
-            //
-            // There isn't much value in complicating code here to return an error from the
-            // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
-            // trouble to other threads/guests or the main() function and should be safe for the
-            // daemon.
-            match cfg {
-                GpioDeviceType::PhysicalDevice(device_num) => {
-                    let controller = PhysDevice::open(device_num).unwrap();
-                    start_device_backend(controller, socket.clone());
-                }
-                #[cfg(feature = "mock_gpio")]
-                GpioDeviceType::SimulatedDevice(num_gpios) => {
-                    let controller = MockGpioDevice::open(num_gpios).unwrap();
-                    start_device_backend(controller, socket.clone());
-                }
-                #[cfg(not(feature = "mock_gpio"))]
-                _ => {
-                    panic!("mock implementatoins aren't supported");
-                }
-            };
-        });
-
-        handles.push(handle);
+        let builder = std::thread::Builder::new();
```

Comment:
This is required for caring about thread launch failures (as in Linux being unable to provision the thread), right? Is there a reason why we need that? I do not see a reference to this in the commit message.

Reply:
Its less about failure to spawn and more about the errors from failures to build the config (CouldNotOpenDevice, CouldNotCreateGpioController, CouldNotCreateBackend) and the final one CouldNotCreateDaemon is actually I think the spawn failure.

Reply:
Hm... Those already happen inside the thread. .spawn() will return a join handle that upon `.join()` returns the `Result` of the thread. The old cold already collects the handles and joins and tries (`??`) both the joinining process and the thread `Result`. So if we do not want to catch the low-level thread spawning errors from the OS, we can keep the old direct `spawn()` I think.

Reply:
OK I've reverted the builder change and minimised the changes to just handling ? properly.